tsdb/engine/tsm1/tombstone.go

```diff
@@ -17,11 +17,17 @@ const (
 )
 
 type Tombstoner struct {
-	mu sync.Mutex
+	mu sync.RWMutex
 
 	// Path is the location of the file to record tombstone. This should be the
 	// full path to a TSM file.
 	Path string
+
+	// cache of the stats for this tombstone
+	fileStats []FileStat
```

Comment:
There's still the concern of the data being modified, though. If we're okay with expecting the result to be used in a read-only fashion, then there are some other changes to allocate the slice exactly once: use `t.fileStats = t.fileStats[:0]` rather than assigning nil, and `t.fileStats = append(t.fileStats[:0], FileStat{ ...` instead of assigning a new slice.